tests/puppeteer/campaigns/functional/BO/13_shopParameters/05_contact/04_sortContacts.js

```diff
@@ -0,0 +1,102 @@
+};
+
+// Sort contacts by id, name, email and description
+describe('Sort Contacts', async () => {
+  // before and after functions
+  before(async function () {
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    this.pageObjects = await init();
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+  });
+
+  // Login into BO and go to contact page
+  loginCommon.loginBO();
+
+  it('should go to \'Shop parameters>Contact\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToContactsPage', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.shopParametersParentLink,
+      this.pageObjects.boBasePage.contactLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
+    const pageTitle = await this.pageObjects.contactsPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.contactsPage.pageTitle);
+  });
+
+  it('should reset all filters and get number of contacts in BO', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'resetFilterFirst', baseContext);
+    numberOfContacts = await this.pageObjects.contactsPage.resetAndGetNumberOfLines();
+    await expect(numberOfContacts).to.be.above(0);
+  });
+
+  // Start sorting contacts
+  const tests = [
+    {
+      args:
+        {
+          testIdentifier: 'sortByIdDesc', sortBy: 'id_contact', sortDirection: 'desc', isFloat: true,
+        },
+    },
+    {args: {testIdentifier: 'sortByNameAsc', sortBy: 'name', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortByNameDesc', sortBy: 'name', sortDirection: 'desc'}},
+    {args: {testIdentifier: 'sortByEmailAsc', sortBy: 'email', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortByEmailDesc', sortBy: 'email', sortDirection: 'desc'}},
+    {args: {testIdentifier: 'sortByDescriptionAsc', sortBy: 'description', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortByDescriptionDesc', sortBy: 'description', sortDirection: 'desc'}},
+    {
+      args:
+        {
+          testIdentifier: 'sortByIdAsc', sortBy: 'id_contact', sortDirection: 'asc', isFloat: true,
+        },
+    },
+  ];
+
+  tests.forEach((test) => {
+    it(`should sort by '${test.args.sortBy}' '${test.args.sortDirection}' And check result`, async function () {
+      await testContext.addContextItem(this, 'testIdentifier', test.args.testIdentifier, baseContext);
+      let nonSortedTable = await this.pageObjects.contactsPage.getAllRowsColumnContent(test.args.sortBy);
+      await this.pageObjects.contactsPage.sortTable(test.args.sortBy, test.args.sortDirection);
+      let sortedTable = await this.pageObjects.contactsPage.getAllRowsColumnContent(test.args.sortBy);
+      if (test.args.isFloat) {
+        nonSortedTable = await nonSortedTable.map(text => parseFloat(text, 10));
+        sortedTable = await sortedTable.map(text => parseFloat(text, 10));
```

Comment:
`parseFloat` doesn't have two parameters : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat

Reply:
Damn, I thought it did, but apparently that's just `parseInt` :o

Reply:
I thought that too, so I deleted the second parameter in all sort tests. thanks @Progi1984